oclint-core/test/RuleBaseTest.cpp

```diff
@@ -0,0 +1,44 @@
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+#include "oclint/RuleBase.h"
+
+using namespace oclint;
+class TestRule : public RuleBase
+{
+public:
+    TestRule(std::string ruleName) : ruleName(ruleName) {}
+    void apply() {}
+    const std::string name() const
+    { return ruleName; }
+    const std::string category() const
+    { return "test"; }
+    int priority() const
+    { return 0; }
+    std::string ruleName;
+};
+
+TEST(RuleBaseTest, EmptyName)
+{
+    EXPECT_EQ(TestRule("").identifier(), "");
+}
+
+TEST(RuleBaseTest, CamelCase)
+{
+    EXPECT_EQ(TestRule("some name").identifier(), "SomeName");
+}
+
+TEST(RuleBaseTest, SpecialChars)
+{
+    EXPECT_EQ(TestRule("some/name?").identifier(), "SomeName");
+}
+
+TEST(RuleBaseTest, AttributeName)
+{
+    EXPECT_EQ(TestRule("attr").attributeName(), "attr");
+}
+
```

Comment:
It might be testing overloading mechanism in C++, but it also serve as an example to show the `identifier` can be overwritten. In addition, just to make sure future changes won't break this.